Sprite.java

```diff
@@ -315,4 +317,53 @@ private void processScripts(JSONArray array)
 
         return;
     }
+
+    /**
+     * Get list of Variables.
+     *
+     * @return variables
+     */
+    public String[] getVariables()
+    {
+        return variables;
+    }
+
+    /**
+     * Populate list of variables.
+     */
+    public void populateVariables()
+    {
+        JSONArray vars = 
+            FileUtils.getJSONArrayAttribute(this.jsonObj, "variables");
+        variables = new String[vars.size()];
+        JSONObject children = new JSONObject();
+        for (int i = 0; i < vars.size(); i++)
+        {
+            children = (JSONObject) vars.get(i);
+            variables[i] = 
+                FileUtils.getJSONAttribute(children, "name");;
+        }
+    }
+
+    /**
+     * Get variable usage count.
+     *
+     * @param var - the variable being counted
+     * @return the number of times the variable is used
+     */
+    public int getVariableUsageCount(String var)
+    {
+        int count = 0;
+        JSONArray scripts =
+            FileUtils.getJSONArrayAttribute(this.jsonObj, "scripts");
```

Comment:
'this' keyword is not necessary here - since we're only referring to a single jsonObj field, just 'jsonObj' is sufficient.